client/app/cloud/project/compute/infrastructure/privateNetwork/cloud-project-compute-infrastructure-privateNetwork-list.directive.js

```diff
@@ -76,6 +89,46 @@ class PrivateNetworkListCtrl {
             .finally(() => this.loaders.vrack.get = false);
     }
 
+
+    /**
+     * open UI activate private network modal if there are pre ordered vRacks else
+     * take user to order new vRack page
+     *
+     * @memberof PrivateNetworkListCtrl
+     */
+    addVRack () {
+        var vRacks = this.vRacks;
+        if(vRacks && _.isArray(vRacks) && !_.isEmpty(vRacks)) {
+                // user has pre ordered vRacks
+                var orderUrl = this.orderUrl;
+                const modal = this.resources.modal.open({
+                    windowTopClass: "cui-modal",
```

Comment:
already done in ControllerModalHelper.showModal

Reply:
fixed